### PR TITLE
Misc minor updates and fixes

### PR DESCRIPTION
**Pull request by roninsightrx**
- misc minor bugfixes in data collection
- adds "middle" (or "cmid") timepoint
- adds "random" timepoint
- adds "scatter=" option to induce random scatter in timepoints

### CODE REVIEW

File: R/create_design.R [hunk 1]
```diff
@@ -12,10 +12,22 @@
...
+#' @param offset offset from standardized PK moments specified in `when`, e.g.
+#' `c(1, -1)` with `when = c("peak", "trough")` to sample 1 hour after peak and
+#' 1 hour before trough.
+#' @param scatter optional random variation in time, specified as the standard
+#' deviation, e.g. `scatter = 0.1` to allow for variation in sampling time with
+#' an SD of 0.1 hours. Only relevant for sampling times, not for
+#' regimen_update designs or target designs. Random variation does not protect
+#' for peaks or troughs becoming sampled during infusion or in previous / next
+#' dose. So value for `scatter` should be chosen appropriately and probably
+#' used in conjunction with approriate `offset` values.
```
Comment (jasmineirx) [inline R/create_design.R:30]:
```suggestion
#' @param scatter optional random variation in time, specified as the standard
#' deviation, e.g. `scatter = 0.1` to allow for variation in sampling time with
#' an SD of 0.1 hours. Only implemented for sampling times (not for
#' regimen_update designs), not applicable for or target designs. Random variation 
#' does not protect for peaks or troughs becoming sampled during infusion or in 
#' previous / next dose. So value for `scatter` should be chosen appropriately and
#' probably used in conjunction with appropriate `offset` values.
```

theoretically we could use this scatter argument to account for non-exact dosing too.

File: R/get_sampling_times_from_schema.R [hunk 2]
```diff
@@ -38,9 +49,13 @@ get_sampling_time_core <- function(row, regimen) {
   if(is.na(t_anchor)) {
     msg <- "Not enough doses in regimen to update, please increase initial regimen length."
     if(row$base %in% c("trough", "cmin")) {
-      msg <- c(msg, " For `cmin` calculation, you will need one more dose in the regimen than the dose for which the `cmin` is calculated.")
+      msg <- c(msg, " For `cmin`, `middle`, or `random` calculation, you will need one more dose in the regimen than the dose for which the `cmin` is calculated.")
```
Comment (jasmineirx) [inline R/get_sampling_times_from_schema.R:52]:
```suggestion
      msg <- c(msg, " For `cmin`, `middle`, `cmid`, or `random` calculation, you will need one more dose in the regimen than the dose for which the `cmin` is calculated.")
```

**Review by jasmineirx — APPROVED** [review]
lgtm

**Comment by roninsightrx** [comment]
> theoretically we could use this scatter argument to account for non-exact dosing too.

Ah yes, good point. Larger change though, could be tackled in different ticket.

**Pull request merged**